index.html

```diff
@@ -1504,6 +1504,47 @@ <h2>Public Keys</h2>
 see Section <a href="#resolution"></a>.
       </p>
 
+<p>
+Revoked keys MUST NOT be present in the <code>publicKey</code> property of a <a>DID document</a>.<br/>
+Revoked keys MAY be present in the <code>revokedKey</code> property of a <a>DID document</a>.
+</p>
+
+<pre class="example nohighlight" title="Revoked public keys">
+{
+  "@context": ["https://www.w3.org/ns/did/v1", "https://w3id.org/security/v1"],
+  "id": "did:example:123456789abcdefghi",
```

Comment:
What happens if a did with "id" = "did:example?123456789abcdefghi" is *also* present in the publicKeys section, with a different value? I believe this would be valid, since current wisdom is to allow a single "id" value to point to more than one value over the lifespan of the DID. But it seems odd.

Reply:
You mean a key with `id = did:example:123#_Qq0UL2Fq651Q0Fjd6TvnYE-faHiOpRlPVQcY_-tA4A` ? `did:example:123456789abcdefghi` is the DID of the example document... While its possible to use it as an identifier, I belive that should not be allowed, and that it is a seperate issue... 

I think you are hinting at 2 potentially problematic things...

1. What if `did:example:123#_Qq0UL2Fq651Q0Fjd6TvnYE-faHiOpRlPVQcY_-tA4A` is in `authentication`, but also in `revokedKey` with values defined there... doesn't that mean that processing of authentication just became more complicated?

2. What if `did:example:123#_Qq0UL2Fq651Q0Fjd6TvnYE-faHiOpRlPVQcY_-tA4A` is in `revokedKeys`, but is also in `publicKey`... because it was replaced with a new key, when it was revoked, but the new key has the same identifier...

I think that case 2 is illegal, because a did document should never have duplicate identifiers... in other words, if you are rotating  something like `did:example:123#primary`, you cannot retain a copy in `revokedKey`...

Potential solution would be to allow "revoked keys" to be in any section, and add language stating that if any key has a "revocation" property, it is considered revoked... The reason I didn't suggest this is that currently absence of a key indicated revocation, and I wanted to keep that property... not add an or presence of key with "revocation" bit to it....

Reply:
Yes, you've correctly articulated my two concerns. (Good job; I was struggling to put it into words.)

I strongly prefer to treat absence of a key as equal to revocation, and not having a "revoked keys" section. Maybe the standard has to allow a "revoked keys" section for some use cases, but I hope I never run into one. So I think we're aligned there.

If you are correct that case 2 is illegal (I hope you are), then this is a pretty concrete reason why the asserted behavior goal of allowing #xyz to mean multiple things over its lifetime is a bad idea (at least, if it refers to a key). Maybe we'll get people to reconsider immutability of IDs in DID docs.

Anyway, I don't have any pushback for you; I'm just pointing out subtleties that could trip us up.

Reply:
My hope was that by disallowing revoked keys to be present in any section but "revokedKeys" we would at least keep them entirely located there.. and for cases where a DID Method chose to support the concept of revoked keys, say for `assertionMethod`, they would implement their verification logic to handle the  lookup for `did:example:123#vc-key` to look first in `publicKey` and second in `revokedKey`, and for everyone else who didn't use revoked keys, nothing would change.

@msporny @dlongley how does this sound to you?

Reply:
-1 to revokedKey (for the reasons stated above). No one is implementing this now, AFAICT.

Reply:
As I have revised this content, I request to resolve this thread to avoid confusion.